src/unittest/test_moveaction.cpp

```diff
@@ -0,0 +1,178 @@
+/*
+Minetest
+Copyright (C) 2021 Minetest core developers & community
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "test.h"
+#include "test_config.h"
+
+#include "mock_inventorymanager.h"
+#include "mock_server.h"
+#include "mock_serveractiveobject.h"
+
+#include <scripting_server.h>
+
+
+class TestMoveAction : public TestBase
+{
+public:
+	TestMoveAction() { TestManager::registerTestModule(this); }
+	const char *getName() { return "TestMoveAction"; }
+
+	void runTests(IGameDef *gamedef);
+
+	void testMove(ServerActiveObject *obj, IGameDef *gamedef);
+	void testMoveSomewhere(ServerActiveObject *obj, IGameDef *gamedef);
+	void testMoveUnallowed(ServerActiveObject *obj, IGameDef *gamedef);
+	void testMovePartial(ServerActiveObject *obj, IGameDef *gamedef);
+	void testSwap(ServerActiveObject *obj, IGameDef *gamedef);
+	void testSwapFromUnallowed(ServerActiveObject *obj, IGameDef *gamedef);
+	void testSwapToUnallowed(ServerActiveObject *obj, IGameDef *gamedef);
+};
+
+static TestMoveAction g_test_instance;
+
+void TestMoveAction::runTests(IGameDef *gamedef)
+{
+	MockServer server;
+
+	g_settings->setBool("secure.enable_security", false);
+	ServerScripting server_scripting(&server);
+	server_scripting.loadScript(std::string(TEST_PATH) + "../../builtin/init.lua");
+	server_scripting.loadScript(std::string(TEST_PATH) + "helper_moveaction.lua");
```

Comment:
Why do you need a separate test path? It would be more fitting to use something like
`porting::path_share + DIR_DELIM "builtin" + DIR_DELIM "init.lua"` in this place.

Reply:
If you move the files to builtin you can probably also avoid toggling mod security.

Reply:
Why can't you?

Reply:
It crashes because `m_modmgr` https://github.com/minetest/minetest/blob/a5f385917d0db5097e3cd4a1c6de794bda20c01f/src/server.cpp#L3659 is `nullptr`.

Reply:
There might be a misunderstanding. I meant to use `Server::getBuiltinLuaPath()` to load those two files, rather than introducing a new testing path. There is really no need to load the entire scripting interface.

It would be somewhat weird if loading files from a different location results in a different behaviour.

Reply:
> If you move the files to builtin you can probably also avoid toggling mod security.
I meant that.


>  meant to use `Server::getBuiltinLuaPath()` to load those two files, rather than introducing a new testing path.
Now it loads everything from `builtin`.